core/src/main/java/hudson/model/Job.java

```diff
@@ -1210,7 +1210,8 @@ public synchronized void doConfigSubmit(StaplerRequest req,
 
             String newName = req.getParameter("name");
             final ProjectNamingStrategy namingStrategy = Jenkins.getInstance().getProjectNamingStrategy();
-            if (newName != null && !newName.equals(name)) {
+            if (newName != null && !newName.trim().equals(name.trim())) {
```

Comment:
@KostyaSha my comment was a suggestion for you to provide insight rather than something which offers no solution. I also agree with @oleg-nenashev , the code seems fine (there's a null check on the incoming data; assume preexisting 'name' is not null), adding anything more will only help to confuse what it's doing.